### PR TITLE
Refreshed Community Sidebar

### PR DESCRIPTION
**Pull request by JakeShirley**
# Checklist
- [x] I have described what this PR contains

**Choose one of the following two options:**
- - [ ] This PR does not introduce major changes
- - [x] This PR introduces major changes, and I have consulted @buresdv, @jboi or @mormaer in the *Mlem Development Matrix room*

**Choose one of the following two options:**
- - [ ] This PR does not change the UI in any way
- - [x] This PR adds new UI elements / changes the UI, and I have attached pictures or videos of the new / changed UI elements

# Pull Request Information

## About this Pull Request
This PR refreshes the sidebar screen with a layout that is more similar to the profile screen mockup.

## Additional Context
n/a

### CODE REVIEW

**Comment by BrooklynMan** [comment]
I like where this is going, but the mod list needs work.

**Comment by JakeShirley** [comment]
Polished the moderator list

image

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Components/Sidebar Header Avatar.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                    image
+                        .resizable()
+                        .clipShape(Circle())
+                        .overlay(Circle()
+                        .stroke(.secondary, lineWidth: 2))
+                } placeholder: {
+                    ProgressView()
+                }
+            } else {
+                // TODO: Default avatar?
```
Comment (mormaer) [inline Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Components/Sidebar Header Avatar.swift:30]: @BrooklynMan - do you know if we have a default avatar we'd like to use here when the user has not set one?

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Components/Sidebar Header Label.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+struct CommunitySidebarHeaderLabel : View {
+    @State var text: String
+    
+    init(_ text: String) {
+        self._text = State(initialValue: text)
+    }
+    
+    var body: some View {
+        Text(text).padding(3).foregroundColor(.white).background(RoundedRectangle(cornerRadius: 5).foregroundColor(.gray)).font(.footnote)
```
Comment (mormaer) [inline Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Components/Sidebar Header Label.swift:18]: just for readability can we multiline these?

```swift
Text(text)
    .padding(3)
    .foregroundColor(...)
    ...
```

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Components/Sidebar Header.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+//  Created by Jake Shirley on 6/21/23.
+//
+
+import SwiftUI
+import CachedAsyncImage
+
+struct CommunitySidebarHeader : View {
+    @State var communityDetails: GetCommunityResponse
+    
+    private func getRelativeTime(date: Date) -> String {
```
Comment (mormaer) [inline Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Components/Sidebar Header.swift:14]: we should look at moving this type of formatting out into extensions or such for use throughout the app before individual functions start cropping up in different views

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Components/Sidebar Header.swift [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                    VStack(alignment: .trailing) {
+                        Spacer().frame(height: 170)
+                        HStack {
+                            CommunitySidebarHeaderLabel("\(communityDetails.communityView.counts.subscribers) Subscribers")
+                            CommunitySidebarHeaderLabel("\(communityDetails.online) Online")
+                        }
+                        Spacer().frame(height: 20)
+                        
+                        Text(communityDetails.communityView.community.name).font(.title).bold().lineLimit(1)
+                        Text("@\(communityDetails.communityView.community.name)@\(communityDetails.communityView.community.actorId.host()!)").font(.footnote).lineLimit(1)
```
Comment (mormaer) [inline Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Components/Sidebar Header.swift:56]: I know the `.host()` shouldn't ever fail here, but can we unwrap?

File: Mlem/Views/Tabs/Posts/Components/Shared/Links/User Profile Label.swift [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    @State var account: SavedAccount
+    @State var user: APIPerson
+    
+    // Extra context about where the link is being displayed
+    // to pick the correct flair
+    @State var postContext: APIPostView? = nil
+    @State var commentContext: APIComment? = nil
+    
+    static let developerNames = [
+        "lemmy.ml/u/lFenix",
```
Comment (mormaer) [inline Mlem/Views/Tabs/Posts/Components/Shared/Links/User Profile Label.swift:23]: I guess we should remove this, as David removed himself from the about page before leaving 😞

**Review by mormaer — CHANGES_REQUESTED** [review]

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Components/Sidebar Header.swift (hunk 3 above)
Comment (JakeShirley) [reply Mlem/Views/Tabs/Posts/Components/Post/Components/Community View/Components/Sidebar Header.swift:14]: Moved it to the extensions group!

File: Mlem/Views/Tabs/Posts/Components/Shared/Links/User Profile Label.swift (hunk 5 above)
Comment (JakeShirley) [reply Mlem/Views/Tabs/Posts/Components/Shared/Links/User Profile Label.swift]: Done

**Review by mormaer — APPROVED** [review]
Looks good to me - I've raised an issue to see if we want to add a default user avatar soon 🤞

**Comment by BrooklynMan** [comment]
I've added some proposals to the default avatar PR #227

**Pull request merged**